pkg/gateway/gateway.go

```diff
@@ -102,7 +102,9 @@ func NewGateway(ctx context.Context, env env.Core, baseLog, accessLog *logrus.En
 			return nil, err
 		}
 
-		domains = append(domains, u.Hostname())
+		if u.Hostname() != "" {
```

Comment:
I like this conditional, but i'm thinking lets revisit the logic. we should probably log or return an error if our hostname is empty, it is a critical error that would result in the gateway not functioning in a way that is expected

Reply:
I think it is hard requirement both ActiveDirectoryEndpoint and ResourceManagerEndpoint hostname is not empty

Reply:
and then adjust tests accordingly, imo gateway is not functioning properly if those fields are missing

Reply:
Going through `environments.go` I see that we have it configured so I also agree if it is missing we should error out. I will make the required changes to the gateway.